src/platform/silabs/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -163,6 +163,8 @@ CHIP_ERROR SlWiFiDriver::ConnectWiFiNetwork(const char * ssid, uint8_t ssidLen,
     wifiConfig.security = WFX_SEC_WPA2;
 
     ChipLogProgress(NetworkProvisioning, "Setting up connection for WiFi SSID: %.*s", static_cast<int>(ssidLen), ssid);
+    // Resetting the retry connection state machine for a new access point connection
+    WifiInterface::GetInstance().ResetScheduledConnectionAttempts();
```

Comment:
Why do we need the reset call here and in the WifiInterface?

Reply:
> What if the connection never succeeds. And later on there is a new connection attempt to a new ssid?
> Looking at the code, it looks like that new connection attempt will also start at whatever the last retryInterval was at.
>
> I think we should do retryInterval = kWlanMinRetryIntervalsInSec; at the start of a new connection attempt and then you don't need to worry about resetting it on a succeeded or failed connection

It is to cover this use case, though currently it is not coming up this way.